lbrynet/lbrynet_daemon/Daemon.py

```diff
@@ -2174,6 +2174,27 @@ def jsonrpc_configure_ui(self, p):
         return d
 
     @AuthJSONRPCServer.auth_required
+    def jsonrpc_open_file(self, p):
+        """
+        Instruct the OS to open a file.
+
+        Args:
+            'path': path of file to be opened
+        Returns:
+            True, opens file
+        """
+        path = p['path']
+        if sys.platform == 'darwin':
+            d = threads.deferToThread(subprocess.Popen, ['open', path])
+        elif os.name == 'posix':
+            d = threads.deferToThread(subprocess.Popen, ['xdg-open', path])
+        elif sys.platform == 'win32':
+            d = threads.deferToThread(os.startfile, path)
```

Comment:
given that this code is nearly identical to `reveal` I'd like there to be some abstraction over the OS for this sort of thing. Its better to do the if statement in a factory that gives you back a class that allows the rest of the code to not care what OS we're on.

roughly

```
def file_factory(path):
    if sys.platform == 'darwin':
        return DarwinFile(path)
    ...

class DarwinFile(object):
    def open(self):
        <code to open>
    def reveal(self):
        <code to reveal>

etc
```